webmin/os-eol-lib.pl

```diff
@@ -0,0 +1,176 @@
+# os-eol-lib.pl
+# Functions for managing OS end-of-life data
+
+use Time::Local;
+
+# eol_oses_list()
+# Returns a list of OSes for which EOL data is available
+sub eol_oses_list
+{
+return ('almalinux', 'amazon-linux', 'centos-stream', 'centos',
+        'debian', 'fedora', 'freebsd', 'openbsd', 'opensuse',
+        'oracle-linux', 'rhel', 'rocky-linux', 'ubuntu');
+}
+
+# eol_get_os()
+# Returns the current OS or undef if the OS is not supported
+sub eol_get_os
+{
+my $os_type = lc($gconfig{'real_os_type'});
+@eol_oses_list = &eol_oses_list();
+my ($os_found) = grep {
+        my $__ = $_;
+        $__ =~ s/-?linux//;
+        $__ =~ s/-/ /;
+        $os_type =~ /$__/ } @eol_oses_list;
+return $os_found if ($os_found);
+return undef;
+}
+
+# eol_fetch_os_data()
+# Fetches the latest EOL data for the current OS and caches it.
+# Returns undef if OS is not supported.
+sub eol_fetch_os_data
+{
+my $os = &eol_get_os();
+return undef if (!$os);
+my ($fetch, $error);
+my $eol_write_cache = sub {
+        my $data = shift;
+        &write_file_contents("$module_var_directory/eolcache", $data);
+};
+&http_download('endoflife.date', 443, "/api/$os.json", \$fetch, \$error, undef, 1);
```

Comment:
Add a timeout to this call

Reply:
The default timeout is 60 seconds. I thought it's fine?

Reply:
Okay, 5 seconds is more than enough.

Reply:
Yeah 5 sewconds is better